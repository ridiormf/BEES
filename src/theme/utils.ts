type Space = number | string;

const utils = {
  marginHorizontal: (value: Space) => ({
    marginLeft: value,
    marginRight: value,
  }),
  marginVertical: (value: Space) => ({
    marginTop: value,
    marginBottom: value,
  }),
  paddingHorizontal: (value: Space) => ({
    paddingLeft: value,
    paddingRight: value,
  }),
  paddingVertical: (value: Space) => ({
    paddingTop: value,
    paddingBottom: value,
  }),
};

export default utils;
