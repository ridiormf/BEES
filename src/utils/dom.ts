export const getBody = () => {
  return document.getElementsByTagName('body')[0];
};
