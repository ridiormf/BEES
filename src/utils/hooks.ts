import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

/**
 * Function that simulates the behavior of a class component constructor,
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component is instantiated.
 */
export const useConstructor = <ValueType = void>(
  callback: () => ValueType,
): ValueType | void => {
  const componentWasBuilt = React.useRef<boolean>(false);
  const value = React.useRef<ValueType>();
  if (!componentWasBuilt.current) {
    componentWasBuilt.current = true;
    value.current = callback();
  }

  return value.current;
};

/**
 * Function that simulates the behavior of a class component method "componentDidMount",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component mount.
 */
export const useDidMount = (callback: () => void): void => {
  React.useEffect(callback, []);
};

/**
 * Function that simulates the behavior of a class component method "componentWillUnmount",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component will unmount.
 */
export const useWillUnmount = (callback: () => void): void => {
  React.useEffect(() => {
    return callback;
  }, []);
};

/**
 * Function that simulates the behavior of a class component method "componentDidUpdate",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called on each value change of variables declared as dependencies,
 * excluding the first render of the component.
 * @param {React.DependencyList} deps Dependency list to be listened.
 */
export const useDidUpdate = (
  callback: () => void,
  deps: React.DependencyList,
): void => {
  const firstRender = React.useRef<boolean>(true);
  React.useEffect(() => {
    if (firstRender.current) {
      firstRender.current = false;
    } else {
      callback();
    }
  }, deps);
};

/**
 * Function that use main hooks from "react-router-dom" to simplify usability.
 *
 * @returns {object} containing following attributes:
 * @type {function} "goBack": to return navigation one page.
 * @type {function} "go": to navigate for path.
 * @type {string} "pathname": current url pathname.
 */
export const useAppNavigation = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const goBack = () => {
    navigate(-1);
  };

  const go = (path: string) => {
    navigate(path);
  };

  return {
    goBack,
    go,
    pathname: location.pathname,
  };
};
