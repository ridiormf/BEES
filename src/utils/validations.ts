/**
 * Function that checks if the full name is valid.
 * It will check:
 * 1. if the string has, at least, two names, and if all names has 2 letters or more;
 * 2. If the string not start or end with blank space;
 * 3. If it doesn't contain numbers; and
 * 4. If it doesn't contain two or more blank spaces between words
 *
 * @param {string} fullName String to be checked.
 * @returns {boolean} Return true if is a valid name, otherwise returns false.
 */
export const checkFullName = (fullName?: string): boolean => {
  if (!fullName) return false;

  const splittedNames = fullName.split(' ');

  return (
    splittedNames.length > 1 &&
    splittedNames.every((name) => name.length > 1 && !/\d/g.test(name))
  );
};
