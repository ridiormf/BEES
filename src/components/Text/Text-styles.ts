import theme from '../../theme';

export const Text = theme.styled('p', {
  fontSize: 14,
  lineHeight: '21px',
  margin: 0,
  fontWeight: 400,
  color: '$primaryText',
});

export const textStyles = {
  styled: {
    Text,
  },
};
