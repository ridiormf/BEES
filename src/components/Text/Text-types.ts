import React from 'react';
import { textStyles } from './Text-styles';

const {
  styled: { Text },
} = textStyles;

export type TextProps = React.ComponentProps<typeof Text>;
