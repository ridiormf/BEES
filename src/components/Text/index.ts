import Text from './Text';
import type { TextProps } from './Text-types';

export { Text };
export type { TextProps };
