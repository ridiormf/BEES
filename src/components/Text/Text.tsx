import React from 'react';

import { textStyles } from './Text-styles';
import { TextProps } from './Text-types';

const {
  styled: { Text: StyledText },
} = textStyles;

const Text: React.FC<TextProps> = ({ ...rest }) => {
  return <StyledText {...rest} />;
};

export default Text;
