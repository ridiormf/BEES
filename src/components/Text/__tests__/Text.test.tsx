import { render } from '@testing-library/react';
import { Wrapper } from '../../../__tests__/utils';
import Text from '../Text';

describe('components ->', () => {
  describe('Text ->', () => {
    describe('snapshots ->', () => {
      it('Should match snapshot', () => {
        const { container } = render(<Text>text</Text>, {
          wrapper: Wrapper,
        });
        expect(container).toMatchSnapshot();
      });
    });
  });
});
