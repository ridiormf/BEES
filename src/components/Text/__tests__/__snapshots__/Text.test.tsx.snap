// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> Text -> snapshots -> Should match snapshot 1`] = `
<div>
  <p
    class="c-caxBfD"
  >
    text
  </p>
</div>
`;
